### PR TITLE
Fix several issues with URL Maps

### LINKED ISSUE #543: [UI] Can't save urlmaps after edit the match pattern
**Describe the bug**
I forked the default profile and now I can't save after editing it.
**To Reproduce**
1. Go to urlmaps
2. Fork the default profile
3. Edit the match rule
4. Try to save

**Expected behavior**
Be able to save

**Screenshots**
Gravar_select-area_20210806112751

**Desktop (please complete the following information):**
 - OS: Manjaro
 - Browser: Opera

**Additional context**
Running on main. I deployed it yesterday

### LINKED ISSUE #544: Weird behaviour when delete a profile of a urlmap
**Describe the bug**
When I click to delete a profile, I don't know what happens, but it goes wrong (and I become unable to save, like in https://github.com/curiefense/curiefense/issues/543)
**To Reproduce**
1. Go to urlmaps
2. Fork some profiles
3. Delete a profile
4. Now the profiles are wrong

**Expected behavior**
Delete the selected profile

**Screenshots**
Gravar_select-area_20210806114818

**Desktop (please complete the following information):**
 - OS: Manjaro
 - Browser: Opera

**Additional context**
Running on main. I deployed it yesterday

### LINKED ISSUE #548: URL maps- can not save after any changes
**Describe the bug**
After changing anything in URL maps save btn is not triggered 
**To Reproduce**
1. Go to URL map and change a path for example - save is not triggered 

**Expected behavior**
should be triggered

### PR DESCRIPTION
**Pull request by Aviv-Galmidi**
* Fix issue where RUL Map would block saving even when valid - Closes #543 Closes #548
* Fix issue where deleting a URL Map entry would forcefully change another entry - Closes #544

Signed-off-by: Aviv.Galmidi <AvivGalmidi@gmail.com>

### CODE REVIEW

File: curiefense/ui/src/doc-editors/URLMapsEditor.vue [hunk 1]
```diff
@@ -591,7 +591,7 @@ export default (Vue as VueConstructor<Vue & {
   watch: {
     selectedDoc: {
       handler: function(val, oldVal) {
-        if (!val || !oldVal || val.id !== oldVal.id) {
+        if (!val || !oldVal || !_.isEqual(val, oldVal)) {
           this.urlMapsDomainMatches()
           this.initialDocDomainMatch = val.match
```
Comment (denis-reblaze) [inline curiefense/ui/src/doc-editors/URLMapsEditor.vue:596]: this will throw an exception if val is falsy

Comment (GabrielLasso) [reply curiefense/ui/src/doc-editors/URLMapsEditor.vue:596]: If val is falsy `!val` will prevent the check from happening

Comment (denis-reblaze) [reply curiefense/ui/src/doc-editors/URLMapsEditor.vue:596]: if val is falsy then all the disjunction `!val ||` is truthy, how does it prevent check from happening?

Comment (GabrielLasso) [reply curiefense/ui/src/doc-editors/URLMapsEditor.vue:596]: `true || foo()` don't call `foo()` iirc. Or I may be wrong, idk

Comment (denis-reblaze) [reply curiefense/ui/src/doc-editors/URLMapsEditor.vue:596]: Ok I got what you mean. It's correct that `true || foo()` doesn't call `foo()` but it calls the code inside the if block. So the line 596 would throw an exception.

Comment (GabrielLasso) [reply curiefense/ui/src/doc-editors/URLMapsEditor.vue:596]: Ohh I see, that is true

File: curiefense/ui/src/doc-editors/__tests__/URLMapsEditor.spec.ts [hunk 2]
```diff
@@ -204,24 +206,86 @@ describe('URLMapsEditor.vue', () => {
...
   afterEach(() => {
-    jest.clearAllMocks();
+    jest.clearAllMocks()
   })
 
-  test('should not send new requests to API if document data updates but document ID does not', async () => {
-    // 4 requests - ACL Policies, WAF Policies, Rate Limits, URL Maps
-    expect(axiosGetSpy).toHaveBeenCalledTimes(4)
+  test('should not send new requests to API if selected branch does not update', async (done) => {
+    jest.resetAllMocks()
```
Comment (denis-reblaze) [inline curiefense/ui/src/doc-editors/__tests__/URLMapsEditor.spec.ts:221]: Is it possible to move all these resetAllMocks to afterEach?

File: curiefense/ui/src/doc-editors/__tests__/URLMapsEditor.spec.ts [hunk 3]
```diff
@@ -204,24 +206,86 @@ describe('URLMapsEditor.vue', () => {
     wrapper = shallowMount(URLMapsEditor, {
       propsData: {
         selectedDoc: urlMapsDocs[0],
-        selectedBranch: 'master',
+        selectedBranch: selectedBranch,
```
Comment (denis-reblaze) [inline curiefense/ui/src/doc-editors/__tests__/URLMapsEditor.spec.ts:209]:
```suggestion
        selectedBranch,
```

**Comment by Aviv-Galmidi** [comment]
Moved to a new PR on `merge` branch

**Pull request closed without merging**